app/services/api/to_link_anchor.rb

```diff
@@ -0,0 +1,43 @@
+include Rails.application.routes.url_helpers
+
+class Api::ToLinkAnchor
+  def initialize(form:)
+    @form = form
+  end
+
+  def call
+    return Failure.new(form.errors) if form.invalid?
+
+    session = ::Session.find_by_url_token(params[:url_token]) or raise NotFound
+    selected_session_ids = [session.id]
+    stream = session.streams.where(sensor_name: params[:sensor_name]).first!
+
+    data =
+    { sensorId: stream.sensor_id,
+      usernames: session.user.username,
+      heat: { highest: stream.threshold_very_high,
+              high: stream.threshold_high,
+              mid: stream.threshold_medium,
+              low: stream.threshold_low,
+              lowest: stream.threshold_very_low },
+    }
+
+    if (session.type == "FixedSession")
```

Comment:
I remember suggesting this already so prolly it's a bad idea. We could have the path generation in both Mobile and Fixed session. That way this if is gone